src/check.ts

```diff
@@ -0,0 +1,66 @@
+import { Rate } from 'k6/metrics'
+
+type FalsyCheckValue =
+    | Array<unknown>
+    | Record<string | number | symbol, unknown>
+    | boolean
+    | string
+    | number
+    | null
+    | undefined
+
+type Checker<T> =
+    | ((value: T) => FalsyCheckValue)
+    | ((value: T) => Promise<FalsyCheckValue>)
+    | FalsyCheckValue
+    | Promise<FalsyCheckValue>
+
+interface CheckMap<T> {
+    [key: string]: Checker<T>
+}
+
+interface SyncCheckMap {
+    [key: string]: ((value: any) => FalsyCheckValue) | FalsyCheckValue
+}
+
+type CheckResult<C extends CheckMap<any>> = C extends SyncCheckMap ? boolean : Promise<boolean>
+
+const checkRate = new Rate('checks')
+
+export function check<C extends CheckMap<any>>(checkers: C): CheckResult<C>
+export function check<T, C extends CheckMap<T>>(value: T, checkers: C): CheckResult<C>
+export function check<T, C extends CheckMap<T>>(value: T | C, checkers?: C): CheckResult<C> {
```

Comment:
Forgive my ignorance as I may have missed something (maybe it's best to add another frontend dev to the PR for approval).

The [original](https://grafana.com/docs/k6/latest/javascript-api/k6/check/) `k6` `check` API can take an optional `tags` parameter, which this new one doesn't seem to. Is it possible to wrap the original `check` API so that it works with async APIs so that we can keep the existing functionality and make maintenance simpler?

Reply:
> maybe it's best to add another frontend dev to the PR for approval

Will do!

> The [original](https://grafana.com/docs/k6/latest/javascript-api/k6/check/) k6 check API can take an optional tags parameter, which this new one doesn't seem to. Is it possible to wrap the original check API so that it works with async APIs so that we can keep the existing functionality and make maintenance simpler?

Somehow I missed that. I will add it.

Reply:
Fixed. 

I had to remove my new shorthand version because it wouldn't work with tags:

```ts
check({ 'is OK': res.status === 200 }, { hello: 'world' })
```

Does the above mean that there's a check called `is OK` with the tag 'hello' or does it mean that it has the value `{ 'is OK': boolean }` and a check called `hello`? It's pretty much impossible to tell. 

I think being a drop-in replacement is more important than adding any improvements.